WcaOnRails/lib/city_validator.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class CityValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return if record.country.nil? || value.blank?
+
+    city_validator = self.class.get_validator_for_country(record.country.iso2)
+    if city_validator
+      reason_why_invalid = city_validator.reason_why_invalid(value)
+      record.errors.add(attribute, reason_why_invalid) if reason_why_invalid
+    end
+  end
+
+  def self.add_validator_for_country(country_iso2, validator)
+    @validators_by_country_iso2 ||= {}
+    @validators_by_country_iso2[country_iso2] = validator
+  end
+
+  def self.get_validator_for_country(country_iso2)
+    @validators_by_country_iso2[country_iso2]
+  end
+end
+
+class CountryCityValidator
+  # A CountryCityValidator has this one method: `reason_why_invalid` that takes
+  # in a city name, and returns a string reason why that name is not valid, or
+  # nil if the name is valid.
+  def reason_why_invalid(city)
+    raise NotImplementedError
+  end
+end
+
+class CityCommaRegionValidator < CountryCityValidator
+  def initialize(type_of_region:, valid_regions:)
+    @type_of_region = type_of_region
+    @valid_regions = valid_regions
+  end
+
+  def reason_why_invalid(city)
+    _city, region = city.split(", ")
+    if region.nil?
+      "is not of the form 'city, #{@type_of_region}'"
```

Comment:
Thought: should we think about I18n here?

Reply:
Ooooh, good point, I totally didn't think about that! Currently, this is only going to be used by delegates, so I have a slight preference for leaving this as is for now, and we can always internationalize it later if we decide we want to?

Another reason I'm a bit loathe to i18n it now is because I think it may change a bit as we add support for new countries, and it would be nice to save our translators some churn.

Definitely willing to i18n this now if you or @viroulep feel differently, though!

Reply:
We have the competition form internationalized, so it would be consistent to do so here as well, but it's not of much importance and I'm all for merging this up and seeing if it does the job as expected first :ship: